python/plugins/processing/gui/wrappers_layout.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    wrappers_layout.py - Layout widget wrappers
+    ---------------------
+    Date                 : February 2019
+    Copyright            : (C) 2019 by fpsampayo
+    Email                : fpsampayo@gmail.com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+
+from qgis.core import QgsProject
+
+from qgis.PyQt.QtWidgets import QComboBox
+
+from processing.gui.wrappers import (
+    BasicWidgetWrapper
+)
+
+
+class LayoutWrapper(BasicWidgetWrapper):
```

Comment:
probably this class is not useful if you use ParameterEnum

Reply:
This class is used to get all the print layouts and use them as parameter. I don't see any other way to do this.

Reply:
Just curious, why is it not possible to use a ParameterEnum and add print layouts in the enum?

Reply:
I'm against using parameter enum here -- I'd MUCH prefer a reusable parameter and widget instead of requiring all layout related algorithms to reimplement the required logic inside an enum.